simpleviewinc/re-theme
# Zen 245 precommit format

**Ticket**: [here](https://jira.simpleviewtools.com/browse/ZEN-245)

## Context

* add a way to run `yarn format` command on pre-commit. if there's any error. abort the commit

## Goal

* on `git commit` it should run the formatting and fixes any syntax stuff if it can before committing.
* if it does not auto fix, it will throw an error and stop the commit

## Updates

* `package.json`
    * added [husky](https://github.com/typicode/husky)
    * added [lint-staged](https://github.com/okonet/lint-staged)

## Testing
* pull branch
* `yarn install`
* modify any file in `/src` by adding something that would trigger an auto fix by `yarn format`
    * example: 
```Javascript
// in /src/constants/constants.js
const a = 
1
console.log(a);
```

* `git add` + `git commit` the file *(do this on the terminal as opposed to vscode if you want to see the logs)*
* verify that it runs `prettier` and `eslint` ONLY on the file that's staged > fixes the syntax and updates the commit

* modify the file again, this time with something that wouldn't get autofixed
   * example:
       * unused var/function
       * single letter variable (other than r/g/b/a)

* `git add` + `git commit` the file
* verify that it throws an error and does NOT commit
